src/conn.rs

```diff
@@ -689,6 +689,15 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
     }
 
     fn on_packet(&mut self, packet: &Packet, now: Instant) {
+        tracing::debug!("Received packet cid={} packetType={} seqNr={} ackNr={} timestamp={} timestampDiff={} remoteWindow={}",
```

Comment:
Did you try using the debug printout `{:?}` of the packet? Seems cleaner than manually extracting each specific field.

Reply:
Because it doesn't look at clean and has not clean
```
Received packet cid=201 packetType=ST_DATA seqNr=25874 ackNr=6553 timestamp=1612960711 timestampDiff=23325 remoteWindow=1048576
```

^ this is what I have now

If I change it to what you are wondering
``tracing::debug!("Received packet {:?}", packet,);``

```
 Received packet Packet { header: PacketHeader { packet_type: Data, version: One, extension: None, conn_id: 201, ts_micros: 1727574141, ts_diff_micros: 33980, window_size: 1048576, seq_num: 61519, ack_num: 20696 }, selective_ack: None, payload: [254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254, 254] }
```
 
 or
 
 ```
 Received packet Packet { header: PacketHeader { packet_type: State, version: One, extension: None, conn_id: 200, ts_micros: 1727643848, ts_diff_micros: 69684, window_size: 1045581, seq_num: 20697, ack_num: 61517 }, selective_ack: None, payload: [] }
 ```
 
 So using ``{:?}`` makes the debug log a lot less readable and also just has a bunch of not needed text like ``Packet { header: PacketHeader {``

Reply:
Also the data log using ``{:?}`` takes up my entire terminal which is unidle state takes up 2 lines

Where mine shown above takes only 1 line

Reply:
So I think the preferred approach here would be to add a custom Debug formatter with these fields, and use it in both directions. Then, it would be useful anywhere that we want to log out a packet's info.

Also, `trace` seems like the appropriate level for a log that prints out every single packet.

Reply:
It is using trace I just cropped that out in my examples since I thought they were distracting for an example context. 

Thank you for the suggestion for the custom debug trait I didn't even think of it thank you 👍

Reply:
Ok I made that adjustment for ``on_packet``, I think the debug in ``on_transmit`` should stay the same though